homeassistant/components/modbus/climate.py

```diff
@@ -89,6 +98,28 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     temp_step = config[CONF_STEP]
     hub_name = config[CONF_HUB]
     hub = hass.data[MODBUS_DOMAIN][hub_name]
+    structure = config[CONF_STRUCTURE]
+
+    if data_type != DATA_TYPE_CUSTOM:
+        try:
+            structure = f">{DEFAULT_STRUCT_FORMAT[data_type][count]}"
+        except KeyError:
+            _LOGGER.error(
+                "Unable to detect data type for %s sensor, try a custom type", name,
```

Comment:
Not very familar with this integration but shouldn't we name that it could not be found because of `count`?
Because invalid `data_type`s are filter via the config validation beforehand.

Reply:
Yes, we can change that error message. I'll do that.